tasks/reserve-crate.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```

Comment:
This script looks a lot like what is done [here](https://github.com/paritytech/release-engineering/blob/master/scripts/release-lib.sh#L1794-L1833) in the lib.

It would be better to avoid duplicating scripts and having those functions reusable as part of the lib.
I understand the publishing is not ready yet but we could pack the following into a similarly named `reserve_crate()`, with the same interface and call it. That will allow swapping thing easier later on.

Reply:
I based this script on what's done for the lib. That repository is private, however, so I'm not sure that we should be pulling its code here.

The script turned out to be different because it is meant to be triggered with inputs from the workflow, so it needs to allow for more customization. Overall it's not complicated, so it could be ported to/from the lib whenever.

Reply:
>  That repository is private, however, so I'm not sure that we should be pulling its code here.

We should not.

> The script turned out to be different because it is meant to be triggered with inputs from the workflow, so it needs to allow for more customization. Overall it's not complicated, so it could be ported to/from the lib whenever.

The changes you made are good and we can bring that to the lib with your customizations. You can even make it easier to use with some defaults such as:
```
author="${CRATE_AUTHOR:-Parity Technologies <admin@parity.io>}"
```
so calling it is even easier (= less input to provide).

The problem with DUP is that the implementations will diverge at some point and it will get messy, thus my proposal to at least converge with the same function name so we can later swap "in place".

The real solution will be to publish part or the whole lib that is currently private so we can reuse the functions of the lib. This is an important point we should talk about.